src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index == size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node node = new Node<>(value, null, null);
+        if (size == 0) {
+            first = node;
+        } else {
+            Node currentNode = last;
+            currentNode.next = node;
+            node.prev = currentNode;
```

Comment:
do you need this line?

Reply:
yes, I update links to the previous and next nodes when add a node to the end of the linked list. Without it a lot of tests fail